tt_metal/distributed/mesh_device.cpp

```diff
@@ -182,28 +210,77 @@ std::vector<chip_id_t> SystemMesh::Impl::get_mapped_physical_device_ids(const Me
                 physical_coordinate,
                 physical_device_id);
         }
+        return physical_device_ids;
+    } 
+    bool requires_rotation = requested_num_rows > system_mesh_rows || requested_num_cols > system_mesh_cols;
+    bool can_rotate = requested_num_rows <= system_mesh_cols && requested_num_cols <= system_mesh_rows;
```

Comment:
Is it possible to `requires_rotation == true` but `can_rotate == false`? What happens then? Is it captured in the the assert on line 227?

Reply:
I've updated this logic to be clearer.